modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/actions/data/sql/client/AbstractSQLAction.java

```diff
@@ -365,14 +365,21 @@ private CallableStatement getPreparedCall(Connection conn, SQLDatasource datasou
     private ArrayList<ColumnDefinition> getColumnDefinitions(ResultSet rs)
             throws SQLException {
         ArrayList<ColumnDefinition> columnDefs = new ArrayList<>();
+        ArrayList<String> columnNames = new ArrayList<>();
         ResultSetMetaData rsMetaData = rs.getMetaData();
         int cols = rsMetaData.getColumnCount();
         for (int i = 1; i <= cols; i++) {
             String colName = rsMetaData.getColumnLabel(i);
+            if (columnNames.contains(colName)) {
+                String tableName = rsMetaData.getTableName(i).toUpperCase();
+                colName = tableName + "." + colName;
+            }
             int colType = rsMetaData.getColumnType(i);
             TypeKind mappedType = SQLDatasourceUtils.getColumnType(colType);
             columnDefs.add(new SQLDataIterator.SQLColumnDefinition(colName, mappedType, colType));
+            columnNames.add(colName);
         }
+        columnNames.clear();
```

Comment:
Why are we clearing this? .. it's not used again. I think we can remove this.

Reply:
Fixed with a56e89440acb2461848b604b4d25ef85f0862843. Thanks